// @ccmos/nestjs-moment/dist/nestjs-moment.cjs.development.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var moment = _interopDefault(require('moment'));
require('moment-timezone');

var MOMENT = 'MomentModuleMoment';
var MOMENT_TZ = 'MomentModuleMomentTZ';
var MOMENT_MODULE_OPTIONS = 'MomentModuleOptions';

var MomentModule = /*#__PURE__*/function () {
  function MomentModule() {}

  MomentModule.forRoot = function forRoot(options) {
    return {
      module: MomentModule,
      global: options.isGlobal || false,
      providers: [{
        provide: MOMENT,
        useValue: function useValue(date) {
          return moment(date);
        }
      }, {
        provide: MOMENT_TZ,
        useValue: function useValue(date, tz) {
          return moment.tz(date, tz || options.tz || '');
        }
      }],
      exports: []
    };
  };

  MomentModule.forRootAsync = function forRootAsync(options) {
    return {
      module: MomentModule,
      global: options.isGlobal || false,
      providers: [{
        provide: MOMENT_MODULE_OPTIONS,
        useFactory: options.useFactory,
        inject: options.inject || []
      }, {
        provide: MOMENT,
        useValue: function useValue(date) {
          return moment(date);
        }
      }, {
        provide: MOMENT_TZ,
        inject: [MOMENT_MODULE_OPTIONS],
        useFactory: function useFactory(options) {
          return function (date, tz) {
            return moment.tz(date, tz || options.tz || '');
          };
        }
      }]
    };
  };

  return MomentModule;
}();

exports.MomentModule = MomentModule;
